### PR TITLE
now only  replacing the first instance of |[a-z] with the single capi…

### PR DESCRIPTION
**Pull request by eeerin**
…tal letter

SP-6236

### CODE REVIEW

**Comment by spacetc62** [comment]
we don't want to only replace the first instance, think:
`flow|driver.workflow|queue.latency`

**Comment by eeerin** [comment]
it's the first instance of |d appearing and then it would do the first instance of |q appearing. before this code was taking the |d and replacing every instance of d with D. that is why. `ts_average(flow|driver.sand*.latency)` turned into `ts_average(flowDriver.sanD*.latency)`

File: pkg/instrumental_adaptor/provider/translator.go [hunk 1]
```diff
@@ -53,14 +53,12 @@ func (t *Translator) GetRespForExternalMetric(response *instrumental.Instrumenta
 
 func getExternalMetricName(metricName string) string {
 	re := regexp.MustCompile(`\|[a-z]?`)
-	var out string
 	ak := re.FindAllStringSubmatch(metricName, -1)
 	for _, v := range ak {
-		n := strings.Replace(v[0], "|", "", -1)
-		metricName = strings.Replace(metricName, n, strings.ToUpper(n), -1)
```
Comment (eeerin) [inline pkg/instrumental_adaptor/provider/translator.go:60]: the -1 here will replace every instance of the string n (which at this point in the original code would have been a single letter character. thus causing the example I showed earlier. but the FindAllStringSubmatch will find each instance that matches the regex even if they are duplicates.

**Comment by spacetc62** [comment]
@eeerin you still want to add a test to this, correct?

**Comment by eeerin** [comment]
yes please. since we have the workaround for the kube files.

File: go.mod [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+	golang.org/x/text v0.3.4
+	golang.org/x/time v0.0.0-20200630173020-3af7569d3a1e
+	google.golang.org/genproto v0.0.0-20201110150050-8816d57aaa9a
+	google.golang.org/grpc v1.27.1
+	gopkg.in/inf.v0 v0.9.1
+	gopkg.in/natefinch/lumberjack.v2 v2.0.0
+	gopkg.in/yaml.v2 v2.2.8
+	k8s.io/api v0.20.5
+	k8s.io/apimachinery v0.20.5
+	k8s.io/apiserver v0.0.0-20180111074221-ad83dacaae87
```
Comment (eeerin) [inline go.mod]: this guy is causing issues. I cannot seem to install this version but if I use v0.20.5 utils/log and some other libs we require from this package no longer exist.

File: pkg/instrumental_adaptor/provider/translator_test.go [hunk 3]
```diff
@@ -15,6 +16,18 @@ import (
...
+	// Create fake instrumentalProvider
+	translator := newFakeTranslator()
+	fmt.Println(translator)
+	originalName = "metric.fake.latency"
+	name = getCustomMetricName(originalName)
+	assert.Equal(t, name, originalName, "Custom metric name should have not changed.")
+
+	metricName = getCustomMetricName("ts_average(flow|driver.sand*.latency)")
+	assert.Equal(t, metricName, "ts_average(flowDriver.sand*.latency)", "Custom metric name should have translated characters to uppercase properly.")
+}
```
Comment (eeerin) [inline pkg/instrumental_adaptor/provider/translator_test.go:29]: i'm not sure if this is correct but I went a head in wrote it.

**Comment by eeerin** [comment]
Closing this cause this repo cannot be rebuilt. We either need to rewrite this repo all together or consider using prometheus?

**Pull request closed without merging**